include/samurai/reconstruction.hpp

```diff
@@ -403,14 +403,27 @@ namespace samurai
         using mesh_id_t = typename mesh_t::mesh_id_t;
         using ca_type   = typename mesh_t::ca_type;
 
+        auto make_field_like = [](std::string const& name, auto& mesh)
+        {
+            if constexpr (Field::is_scalar)
+            {
+                return make_scalar_field<typename Field::value_type>(name, mesh);
+            }
+            else
+            {
+                return make_vector_field<typename Field::value_type, Field::n_comp, detail::is_soa_v<Field>>(name, mesh);
+            }
+        };
+
```

Comment:
Can we add this in another place ? It will probably useful to have it. Something similar to `zeros_like`, `empty_like`, ...

Reply:
We only need it here...

Reply:
I have other examples where we need to have this kind of helper functions.
Could you open an issue that mentions this code and asks to implement these helper functions ?

Reply:
[[Feat]: initialize a new field with same type of on other #305](https://github.com/hpc-maths/samurai/issues/305)